pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -39,7 +45,49 @@ type openShiftClusterDynamicValidator struct {
 
 	oc              *api.OpenShiftCluster
 	subscriptionDoc *api.SubscriptionDocument
-	fpAuthorizer    refreshable.Authorizer
+	fpAuthorizer    autorest.Authorizer
+}
+
+func ensureAccessTokenClaims(ctx context.Context, tokenCredential *azidentity.ClientSecretCredential, scopes []string) error {
```

Comment:
Q: Why is this necessary now when it wasn't before? It makes sense to validate the FPAuthorizer generates tokens with proper scope, but curious how this impacts downstream code.

Reply:
I probably should have added a comment there.

I copied that logic from the old [GetToken function in pkg/util/aad/aad.go](https://github.com/Azure/ARO-RP/blob/f6e624fcd928468b8208e825bde841a89bcb8303/pkg/util/aad/aad.go#L49-L86) before I deleted the file.  This was the one place that seemed like the token polling logic might still be needed, based on the comment
```
// during credentials rotation this can take time to propagate
```